src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,113 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private static final int LIST_MAX_SIZE = Integer.MAX_VALUE - 8;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_LENGTH];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            expandArray();
+        }
+        array[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size += 1;
+        if (array.length <= size) {
+            expandArray();
+        }
+        indexValidation(index);
+        T[] newArray = Arrays.copyOf(array, array.length);
```

Comment:
you don't need this copy, you can use the same array in System.arraycopy()

Reply:
Did